docs/builtin-tools.md

```diff
@@ -566,6 +567,93 @@ _(This example is complete, it can be run "as is")_
 | `headers`             | ✅ | ❌ |
 
+## File Search Tool
+
+The [`FileSearchTool`][pydantic_ai.builtin_tools.FileSearchTool] enables your agent to search through uploaded files using vector search, providing a fully managed Retrieval-Augmented Generation (RAG) system. This tool handles file storage, chunking, embedding generation, and context injection into prompts.
+
+### Provider Support
+
+| Provider | Supported | Notes |
+|----------|-----------|-------|
+| OpenAI Responses | ✅ | Full feature support. Requires files to be uploaded to vector stores via the [OpenAI Files API](https://platform.openai.com/docs/api-reference/files). Vector stores must be created and file IDs added before using the tool. |
+| Google (Gemini) | ✅ | Requires files to be uploaded via the [Gemini Files API](https://ai.google.dev/gemini-api/docs/files). Files are automatically deleted after 48 hours. Supports up to 2 GB per file and 20 GB per project. Using built-in tools and function tools (including [output tools](output.md#tool-output)) at the same time is not supported; to use structured output, use [`PromptedOutput`](output.md#prompted-output) instead. |
+| Anthropic | ❌ | Not supported |
+| Groq | ❌ | Not supported |
+| OpenAI Chat Completions | ❌ | Not supported |
+| Bedrock | ❌ | Not supported |
+| Mistral | ❌ | Not supported |
+| Cohere | ❌ | Not supported |
+| HuggingFace | ❌ | Not supported |
+| Outlines | ❌ | Not supported |
+
+### Usage
+
+#### OpenAI Responses
+
+With OpenAI, you need to first upload files to a vector store, then reference the vector store IDs when using the `FileSearchTool`:
+
+```py {title="file_search_openai.py" test="skip"}
+from pydantic_ai import Agent, FileSearchTool
+
+agent = Agent(
+    'openai-responses:gpt-5',
+    builtin_tools=[FileSearchTool(vector_store_ids=['vs_abc123'])]  # (1)
+)
+
+result = agent.run_sync('What information is in my documents about pydantic?')
+print(result.output)
+#> Based on your documents, Pydantic is a data validation library for Python...
+```
+
+1. Replace `vs_abc123` with your actual vector store ID from the OpenAI API.
+
+#### Google (Gemini)
+
+With Gemini, you need to first upload files via the Files API, then reference the file resource names:
+
+```py {title="file_search_google.py" test="skip"}
+from pydantic_ai import Agent, FileSearchTool
+
+agent = Agent(
+    'google-gla:gemini-2.5-flash',
+    builtin_tools=[FileSearchTool(vector_store_ids=['files/abc123'])]  # (1)
+)
+
+result = agent.run_sync('Summarize the key points from my uploaded documents.')
+print(result.output)
+#> The documents discuss the following key points: ...
+```
+
+1. Replace `files/abc123` with your actual file resource name from the Gemini Files API.
+
+!!! note "Gemini File Search API Status"
+    The File Search Tool for Gemini was announced on November 6, 2025. The implementation may require adjustment as the official `google-genai` SDK is updated to fully support this feature.
+
+### Configuration
```

Comment:
I think we can drop this section as it's effectively covered by the examples further up. We can add a section once we have optional config options.